demisto_sdk/commands/pre_commit/pre_commit_command.py

```diff
@@ -0,0 +1,247 @@
+import logging
+import multiprocessing
+import os
+import re
+import shutil
+import subprocess
+from collections import defaultdict
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Dict, Iterable, List, Optional, Set
+
+import more_itertools
+from packaging.version import Version
+
+from demisto_sdk.commands.common.constants import INTEGRATIONS_DIR, SCRIPTS_DIR
+from demisto_sdk.commands.common.content_constant_paths import CONTENT_PATH, PYTHONPATH
+from demisto_sdk.commands.common.git_util import GitUtil
+from demisto_sdk.commands.common.handlers import JSON_Handler, YAML_Handler
+from demisto_sdk.commands.content_graph.objects.base_content import BaseContent
+from demisto_sdk.commands.content_graph.objects.integration_script import (
+    IntegrationScript,
+)
+from demisto_sdk.commands.pre_commit.hooks.mypy import MypyHook
+from demisto_sdk.commands.pre_commit.hooks.pycln import PyclnHook
+from demisto_sdk.commands.pre_commit.hooks.ruff import RuffHook
+
+logger = logging.getLogger("demisto-sdk")
+yaml = YAML_Handler()
+json = JSON_Handler()
+
+GITHUB_ACTIONS = os.getenv("GITHUB_ACTIONS")
+DEFAULT_PYTHON_VERSION = "3.10"
+EMPTY_PYTHON_VERSION = "2.7"
+
+PRECOMMIT_TEMPLATE_PATH = Path(__file__).parent / ".pre-commit-config_template.yaml"
+
+SKIPPED_HOOKS = ("format", "validate")
+
+INTEGRATION_SCRIPT_REGEX = re.compile(r"^Packs/.*/(?:Integrations|Scripts)/.*.yml$")
+
+
+with open(PRECOMMIT_TEMPLATE_PATH) as f:
+    PRECOMMIT_TEMPLATE = yaml.load(f)
+
+
+@dataclass
+class PreCommitRunner:
+    """This class is responsible of running pre-commit hooks."""
+
+    python_version_to_files: Dict[str, Set[Path]]
+
+    def __post_init__(self):
+        """
+        We initialize the hooks and all files for later use.
+        """
+        self.hooks = {}
+        for repo in PRECOMMIT_TEMPLATE["repos"]:
+            for hook in repo["hooks"]:
+                self.hooks[hook["id"]] = hook
+        self.all_files: Set[Path] = set()
+        for _, files in self.python_version_to_files.items():
+            self.all_files |= files
+
+    def run(
+        self,
+        test: bool = False,
+        skip_hooks: Optional[List[str]] = None,
+        force_run_hooks: Optional[List[str]] = None,
```

Comment:
perhaps `hooks` shouldn't be strings, but enums?

Reply:
The issue here that we will have to maintain the enum class and the template.
When adding/removing/renaming a hook, we will need to modify in two places instead of one. Need to decide about this.

Reply:
enums are safer than strings anyway, worth the two places imo, but I won't insist